### ISSUE TITLE
Order of test execution

### ISSUE THREAD

**Issue by gjuillot** [testng-team/testng#1122]
Hi,

I got a very smple Maven project with the following two test classes

``` java
public class ATest {

  @BeforeSuite
  public void setPreserveOrder(ITestContext context) {
    System.out.println("context.getCurrentXmlTest().getPreserveOrder() = " + context.getCurrentXmlTest().getPreserveOrder());
  }

  @Test
  public void testA1(ITestContext context) {
    System.out.println("A1");
  }

  @Test(dependsOnMethods = "testA1")
  public void testA2() {
    System.out.println("A2");
  }
}
```

and

``` java
public class BTest {

  @Test
  public void testB1() {
    System.out.println("B1");
  }

  @Test(dependsOnMethods = "testB1")
  public void testB2() {
    System.out.println("B2");
  }
}
```

Given that according to documentation preserve-order is set to true by default, I expected to have

> context.getCurrentXmlTest().getPreserveOrder() = true
> A1
> A2
> B1
> B2

but I got (whatever TestNG version I try)

> context.getCurrentXmlTest().getPreserveOrder() = false
> A1
> B1
> A2
> B2

Removing dependencies between tests, I have the expected sequence A1, A2, B1, B2.
Having in `@BeforeSuite` the `context.getCurrentXmlTest().setPreserveOrder("true")` also make the sequence I expect.

**Comment by juherr** [comment]
Thanks for the report.

Maybe related to #288 and/or #271

What is very strange is the result of `context.getCurrentXmlTest().getPreserveOrder() = false` which is supposed to be `true` by default.

**Comment by gjuillot** [comment]
I agree that default value of preserverOrder seems wrong.

Digging in the question, there's no default value of group-by-instances mentionned in the documentation despite, setting it to true in before suite solves also the issue (except that BTest plays before ATest but that would be okay for me)

**Comment by juherr** [comment]
The default value for preserve-order is located here: https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/xml/XmlSuite.java#L193

But for an unknown reason, it is not used here:
https://github.com/cbeust/testng/blob/master/src/main/java/org/testng/TestNG.java#L852

**Comment by gjuillot** [comment]
Issue indeed solved with 6.9.13-SNAPSHOT. Thank you @juherr and @cbeust
